### ISSUE TITLE
Remove dependency on pillow (PIL)

### ISSUE THREAD

**Issue by gustavla** [apple/turicreate#380]
Pillow is one of Turi Create's hard dependencies. However, it is only used in two situations:

- For unit tests to generate random `tc.Image` images. If we made it easy to take a numpy array and turn it into a `tc.Image`, this need for PIL would go away.
- When calling `image.pixel_data`. Since we have our own bindings to decoding libraries, we should just use them instead of using PIL here.

These two cases can easily be removed. The second also leads to the weird situation where sometimes decoding is done with PIL and sometimes with our own decoder:

```python
# Decodes using PIL
x = image.pixel_data

# Decodes using Turi Create (although the decoded image still passes through PIL)
tc.image_analysis.resize(image, image.width, image.height, image.channels, 
                         decode=True).pixel_data
```

Making this more consistent is partly motivated by #375 and #77, since one solution there is to convert the <8 bit depths to 8 bit in the decoding process. However, this can easily expose the fact that there are two underlying decoding methods, when inconsistencies start appearing.

Note, the main motivation is not to get rid of a dependency (that's just a bonus) and this does not preclude future usages of PIL in Turi Create, if that's ever useful. However, in those cases, it is probably better to make pillow an optional dependency.

**Comment by TobyRoseman** [comment]
@gustavla - this sounds great. 

Regarding the first situation - having an easy way to turn a numpy array into a `tc.Image` would be useful. However it's easier to just make pillow a dependency for running our units test rather than a dependency for Turi Create. You should just need to add it to [this file](https://github.com/apple/turicreate/blob/15ef2d84ee554e446a2a1cc1057d2360672520fc/scripts/requirements.txt)

**Comment by gustavla** [comment]
@TobyRosemen Good point! I'll start with that and then we can do numpy -> `tc.Image` separately.
